libpod/runtime_pod_linux.go

```diff
@@ -220,56 +209,77 @@ func (r *Runtime) removePod(ctx context.Context, p *Pod, removeCtrs, force bool)
 		}
 	}
 
-	// Start removing containers
-	// We can remove containers even if they have dependencies now
-	// As we have guaranteed their dependencies are in the pod
+	// Remove all containers in the pod from the state.
+	if err := r.state.RemovePodContainers(p); err != nil {
+		return err
+	}
+
+	var removalErr error
+
+	// Clean up after our removed containers.
+	// Errors here are nonfatal - the containers have already been evicted.
+	// We'll do our best to clean up after them, but we have to keep going
+	// and remove the pod as well.
+	// From here until we remove the pod from the state, no error returns.
 	for _, ctr := range ctrs {
+		// The container no longer exists in the state, mark invalid.
+		ctr.valid = false
+
+		ctr.newContainerEvent(events.Remove)
+
 		// Clean up network namespace, cgroups, mounts
 		if err := ctr.cleanup(ctx); err != nil {
-			return err
+			if removalErr == nil {
+				removalErr = err
```

Comment:
I think it would be good to add a logrus.debug here.  Might make it easier later on to track down exactly what went south.  Ditto the other "removalErr = err" lines below.

Reply:
I think most of these should already be good errors including the container ID, but I'll slap some Wrapfs on them to be sure

Reply:
Alright, I checked, and everything not already wrapped is returning a well-formatted, wrapped error

Reply:
@mheon Thanks!

Reply:
And the errors are clear enough to know that it came from ctr.cleanup failing at line 231 vs ctr.teardownStorage failing at line 240?

Reply:
If you grep for the specific strings we wrap with, yes - you'll be able to get the specific function that failed

Reply:
OK then, I'll trust you on this one @mheon, thanks for the discussion.